src/workerd/io/tracer.c++

```diff
@@ -440,6 +437,27 @@ void WorkerTracer::recordTimestamp(kj::Date timestamp) {
   }
 }
 
+kj::Date BaseTracer::getTime() {
+  auto& weakIoCtx = KJ_ASSERT_NONNULL(weakIoContext);
+  kj::Date timestamp = kj::UNIX_EPOCH;
+  weakIoCtx->runIfAlive([&timestamp](IoContext& context) { timestamp = context.now(); });
```

Comment:
You'll notice that this is partially redundant with adjustSpanTime – I think we should try to clean that up eventually but currently this is difficult as we are reporting different error messages and have a good fallback option for span tracing (start time) but not for return reporting.